crypto/src/address/r1cs.rs

```diff
@@ -0,0 +1,76 @@
+use crate::Address;
+use ark_ff::PrimeField;
+use ark_r1cs_std::prelude::*;
+use ark_relations::r1cs::SynthesisError;
+use decaf377::{
+    r1cs::{ElementVar, FqVar},
+    Element, Fq,
+};
+
+pub struct AddressVar {
+    pub diversified_generator: ElementVar,
+    pub transmission_key: ElementVar,
+    pub transmission_key_s: FqVar,
+    pub clue_key: FqVar,
+}
+
+impl AddressVar {
+    pub fn diversified_generator(&self) -> ElementVar {
+        self.diversified_generator.clone()
+    }
+
+    #[allow(dead_code)]
+    pub fn transmission_key(&self) -> ElementVar {
+        self.transmission_key.clone()
+    }
+
+    pub fn transmission_key_s(&self) -> FqVar {
+        self.transmission_key_s.clone()
+    }
```

Comment:
Why do we want both of these? Since the `ElementVar` handles cached encodings internally now, could we just use `transmission_key()` everywhere?